test/saithrift/tests/sail2.py

```diff
@@ -757,3 +757,570 @@ def runTest(self):
+            self.client.sai_thrift_remove_vlan_member(vlan_member1)
+            self.client.sai_thrift_remove_vlan_member(vlan_member2)
+            self.client.sai_thrift_remove_vlan_member(vlan_member3)
+
+            ### remove valn
+            self.client.sai_thrift_remove_vlan(vlan_oid)
+
+################################################################
+###        L2MacMoveTest-II (With IPv4 Neighbor entry)       ###
+################################################################
+@group('l2')
+class L2MacMoveTestII (sai_base_test.ThriftInterfaceDataPlane):
+    def runTest(self):
+        '''
+        Description:
+        Create a neighbor entry with MAC1. Let MAC1 be learnt on port 1 and observe traffic
+        forwarding. Execute "L2MacMoveTest" and let MAC1 be learnt on 2nd port. Verify L3
+        forwarding and ensure that traffic is now forwarded to the newly learnt port.
+
+        Steps:
+        1. Create VLAN 10 and associate 2 untagged ports (port 1,2) as the member port of VLAN 10.
+        2. Set attribute value of "Port VLAN ID 10" for the ports 1, 2
+        3. Create Virtual router V1 and enable V4.
+        4. Create  virtual router interfaces and set the interface type as VLAN for VLAN 10
+           and create one more port3 (RIF Id2).
+        5. Create IPv6 neighbor entry with MAC1 and asociate with "RIF Id1".
+        6. Send test packet from port 1 with src_mac = MAC1 address.
+        7. Send IP packet from port 3 with dst_mac= MAC1 and observe traffic forwarding to port 1.
+        8. Send IP packet from port 2 with src_mac = MAC1 and verify the FDB entry, which have learnt via Port 2.
+        9. Send again the test packet from port 3 with dst_mac = MAC1 address and verify it
+            forward to port 2 and not port 1.
+        10. Remove the router interface and change the port attribute to default VLAN.
+        11. Remove the vlan members and VLAN 10 from the database.
+        '''
+
+        switch_init(self.client)
+        vlan_id = 10
+        port1 = port_list[0]
+        port2 = port_list[1]
+        port3 = port_list[2]
+        port4 = port_list[3]
+        dmac1 = '00:33:33:33:33:33'
+        dmac2 = '00:44:44:44:44:44'
+        v4_enabled = 1
+        v6_enabled = 0
+        mac = ''
+        addr_family = SAI_IP_ADDR_FAMILY_IPV4
+        ip_addr1 = '192.168.0.1'
+        ip_addr2 = '10.10.10.1'
+        ip_addr1_subnet = '192.168.0.0'
+        ip_mask1 = '255.255.255.0'
+
+        ### create vlan
+        vlan_oid = sai_thrift_create_vlan(self.client, vlan_id)
+
+        ### create vlan membership with ports
+        vlan_member1 = sai_thrift_create_vlan_member(self.client, vlan_oid, port1, SAI_VLAN_TAGGING_MODE_UNTAGGED)
+        vlan_member2 = sai_thrift_create_vlan_member(self.client, vlan_oid, port2, SAI_VLAN_TAGGING_MODE_UNTAGGED)
+        vlan_member3 = sai_thrift_create_vlan_member(self.client, vlan_oid, port3, SAI_VLAN_TAGGING_MODE_UNTAGGED)
```

Comment:
according to above description, port 3 is not in the vlan 10 but is a port-based router interface. can you correct that?

Reply:
corrected description,  in accordance with script